src/pages/Search/__test__/Search.test.js

```diff
@@ -0,0 +1,95 @@
+import React from 'react';
+import { Route, MemoryRouter } from 'react-router-dom';
+import {
+  render,
+  cleanup,
+  waitForElementToBeRemoved,
+  fireEvent,
+  waitFor,
+} from '@testing-library/react';
+import '@testing-library/jest-dom/extend-expect';
+import { ThemeProvider } from 'styled-components';
+import axiosMock from 'axios';
+import Search from '../index';
+import Theme from '../../../styles/theme';
+import posts from './posts.json';
+
+jest.mock('axios');
+afterEach(cleanup);
+
+const renderWithRouter = (component, subreddit) => ({
+  ...render(
+    <ThemeProvider theme={Theme}>
+      <MemoryRouter initialEntries={[`/search/${subreddit}`]}>
+        <Route path="/search/:subreddit">{component}</Route>
+      </MemoryRouter>
+    </ThemeProvider>,
+  ),
+});
+
+it('Should show javascript on search input', async () => {
+  const { getByTestId } = renderWithRouter(<Search />, 'javascript');
+
+  const input = getByTestId('search-input');
+
+  expect(input.value).toBe('javascript');
+
+  await waitForElementToBeRemoved(() => getByTestId('loader'));
+});
+
+it('Should load data and show heatmap', async () => {
+  axiosMock.get.mockResolvedValueOnce({
+    data: posts,
+  });
+
+  const { getByTestId } = renderWithRouter(<Search />, 'javascript');
+
+  const heatmap = await waitFor(() => getByTestId('heatmap'));
+
+  expect(heatmap).toHaveTextContent('All times are shown in your timezone');
+  expect(axiosMock.get).toHaveBeenCalledTimes(2);
+});
+
+it('Should update url on button click and load data', async () => {
+  axiosMock.get.mockResolvedValue({
+    data: posts,
+  });
+
+  const { getByTestId } = renderWithRouter(<Search />, 'javascript');
+
+  const input = await getByTestId('search-input');
+  fireEvent.change(input, { target: { value: 'reactjs' } });
+  expect(input.value).toBe('reactjs');
+  fireEvent.click(getByTestId('search-button'));
+
+  const heatmap = await waitFor(() => getByTestId('heatmap'));
```

Comment:
Instead of this you can write

```
const heatmap = await findByTestId('heatmap'); 
```

Reply:
if I removed waitFor(), It gives me an error` Unable to find an element by: [data-testid="heatmap"]` and `An update to SearchPage inside a test was not wrapped in act(...)`

Reply:
You need to use `findByTestId` instead of `getByTestId`. `findBy...` is waiting for the element to appear. That's also why it's async. `getBy...` is synchronous and check if the element is there at that point in time, meaning it doesn't wait

Reply:
hahaha sorry, I did not see I thought it was `getByTestId`